haystack/telemetry_2.py

```diff
@@ -185,26 +190,26 @@ def send_pipeline_run_event(  # type: ignore
             event_properties["pipeline.run_parameters.params"] = bool(params)
             event_properties["pipeline.run_parameters.debug"] = bool(debug)
 
-            telemetry.send_event(event_name=event_name, event_properties=event_properties)
+            telemetry.send_event(event_name="Pipeline run", event_properties=event_properties)
     except Exception as e:
         # Never let telemetry break things
-        logger.debug("There was an issue sending a %s telemetry event", event_name, exc_info=e)
+        logger.debug("There was an issue sending a '%s' telemetry event", function_name, exc_info=e)
 
 
-def send_pipeline_event(pipeline: "Pipeline", event_name: str):  # type: ignore
+def send_pipeline_event(pipeline: "Pipeline", event_name: str, event_properties: Optional[Dict[str, Any]] = None):  # type: ignore
     """
     Send a telemetry event related to a pipeline which is not a call to run(), if telemetry is enabled.
     """
     try:
         if telemetry:
-            telemetry.send_event(
-                event_name=event_name,
-                event_properties={
+            if event_properties:
+                event_properties = {
+                    **event_properties,
```

Comment:
nit: since you reassign `event_properties` I'd use `update` instead:
```py
event_properties.update({
    "pipeline.classname": pipeline.__class__.__name__,
    "pipeline.fingerprint": pipeline.fingerprint,
    "pipeline.yaml_hash": pipeline.yaml_hash
})
```